lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -337,9 +338,11 @@ function (Query\Parameter $parameter) use ($key) : bool {
     /**
      * Sets a collection of query parameters.
      *
-     * @param ArrayCollection|mixed[] $parameters
+     * @param ArrayCollection|Parameter[] $parameters
```

Comment:
This does not look right… on line 354, `$value` is could be a `Parameter` instance if I look at the right hand part of this phpdoc type, and I don't think `new Parameter($key, $value)` makes sense in that case.

Reply:
You're right. I was confused by `ArrayCollection|mixed[]` that I assumed was a way to express `ArrayCollection<mixed>` and then, I saw that it was assigned to `$this->parameters` who is indeed an `ArrayCollection<Parameter>`. In fact, the param is `ArrayCollection<Parameter>|mixed[]`. I'll fix this